### PR TITLE
Expand details in missing signatures error

### PR DESCRIPTION
**Pull request by rnicoll**
I'd appreciate thoughts on what the notary should be throwing, as I'm getting less convinced I'm even throwing an appropriate exception. Maybe the exception should simply wrap the cause instead of unpacking/repacking. Thoughts?

### CODE REVIEW

**Comment by exFalso** [comment]
Yes, simply putting the underlying exception as cause sounds good to me

**Comment by rnicoll** [comment]
Now just passes the whole exception around

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt [hunk 1]
```diff
@@ -182,9 +184,10 @@ sealed class NotaryError {
     /** Thrown if the time specified in the timestamp command is outside the allowed tolerance */
     class TimestampInvalid : NotaryError()
 
-    class TransactionInvalid : NotaryError()
+    class TransactionInvalid(val msg: String) : NotaryError()
+    class SignaturesInvalid(val msg: String): NotaryError()
 
-    class SignaturesMissing(val missingSigners: Set<CompositeKey>) : NotaryError() {
-        override fun toString() = "Missing signatures from: ${missingSigners.map { it.toBase58String() }}"
+    class SignaturesMissing(val cause: SignedTransaction.SignaturesMissingException) : NotaryError() {
```
Comment (exFalso) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: There is already a built-int `cause` field in Java exceptions, we should use that

File: node/src/test/kotlin/net/corda/node/services/statemachine/StateMachineManagerTests.kt [hunk 2]
```diff
@@ -319,33 +320,37 @@ class StateMachineManagerTests {
                     node2.info.legalIdentity)))
             net.runNetwork()
         }
+        val endpoint = net.messagingNetwork.endpoint(notary1.net.myAddress as InMemoryMessagingNetwork.PeerHandle)!!
+        val notary1Address: MessageRecipients = endpoint.getAddressOfParty(notary1.services.networkMapCache.getPartyInfo(notary1.info.notaryIdentity)!!)
```
Comment (exFalso) [inline node/src/test/kotlin/net/corda/node/services/statemachine/StateMachineManagerTests.kt]: Needs a rebase onto master

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt (hunk 1 above)
Comment (rnicoll) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: `NotaryError` isn't an exception, I'm just making it look like one

**Comment by rnicoll** [comment]
Rebased

**Pull request merged**